test/segment-loader.test.js

```diff
@@ -603,5 +603,125 @@ QUnit.module('SegmentLoader', function(hooks) {
       assert.ok(!playlistUpdated.segments[0].end,
                 'no end info for first segment of new playlist');
     });
+
+    QUnit.test('errors when trying to switch from audio and video to audio only',
+    function(assert) {
+      const playlist = playlistWithDuration(40);
+      const errors = [];
+
+      loader.on('error', () => errors.push(loader.error()));
+
+      loader.playlist(playlist);
+      loader.mimeType(this.mimeType);
+      loader.load();
+      this.clock.tick(1);
+      loader.syncController_.probeSegmentInfo = () => {
+        return {
+          start: 0,
+          end: 10,
+          source: 'video'
+        };
+      };
+      this.requests[0].response = new Uint8Array(10).buffer;
+      this.requests.shift().respond(200, null, '');
+      loader.buffered_ = () => videojs.createTimeRanges([[0, 10]]);
+      this.updateend();
+      this.clock.tick(1);
+
+      assert.equal(errors.length, 0, 'no errors');
+
+      loader.syncController_.probeSegmentInfo = () => {
+        return {
+          start: 10,
+          end: 20,
+          source: 'audio'
+        };
+      };
+      this.requests[0].response = new Uint8Array(10).buffer;
+      this.requests.shift().respond(200, null, '');
+
+      assert.equal(errors.length, 1, 'one error');
+      assert.equal(errors[0].message,
+                   'Only audio in segment when we expected video (this could be due to' +
+                   ' a lack of codecs specified in the manifest).',
+                   'correct error message');
+    });
+
+    QUnit.test('does not error when audio only', function(assert) {
+      const playlist = playlistWithDuration(40);
+      const errors = [];
+
+      loader.on('error', () => errors.push(loader.error()));
+
+      loader.playlist(playlist);
+      loader.mimeType(this.mimeType);
+      loader.load();
+      this.clock.tick(1);
+      loader.syncController_.probeSegmentInfo = () => {
+        return {
+          start: 0,
+          end: 10,
+          source: 'audio'
+        };
+      };
+      this.requests[0].response = new Uint8Array(10).buffer;
+      this.requests.shift().respond(200, null, '');
+      loader.buffered_ = () => videojs.createTimeRanges([[0, 10]]);
+      this.updateend();
+      this.clock.tick(1);
+
+      assert.equal(errors.length, 0, 'no errors');
+
+      loader.syncController_.probeSegmentInfo = () => {
+        return {
+          start: 10,
+          end: 20,
+          source: 'audio'
+        };
+      };
+      this.requests[0].response = new Uint8Array(10).buffer;
+      this.requests.shift().respond(200, null, '');
+
+      assert.equal(errors.length, 0, 'no errors');
+    });
+
+    QUnit.test('does not error when going from audio only to audio and video',
```

Comment:
Do we support this?

Reply:
If I switch from audio only to audio/video rendition it gives me error as VIDEOJS: ERROR: DOMException: Failed to set the 'duration' property on 'MediaSource': The 'updating' attribute is true on one or more of this MediaSource's SourceBuffers.

Reply:
I'm not sure if we support it, but I put the test in since we don't want to throw the error there for now (until we re-consider). I can remove the test case if we want, until we are sure about behavior.

Seems from @shahlabs ' testing we may want to consider an error anyway.

Reply:
Well according to the MSE spec, going from audio only to audio + video will be dependent on browser

>For example, a user agent may throw a QuotaExceededError exception if the media element has reached the HAVE_METADATA readyState. This can occur if the user agent's media engine does not support adding more tracks during playback.
https://w3c.github.io/media-source/#dom-mediasource-addsourcebuffer

So if we want to support this, we should make sure which browsers support it, and handle the error in browsers that dont. I think its preferable to just not allow switching between audio only and video+audio and vice versa.

Might be worth seeing what happens when going from video only to audio+video as well, though I'm not sure if that is something anyone is trying to do in their manifests.